### PR TITLE
[docs] Add more examples to `#in_order_of` [ci-skip]

### PR DESCRIPTION
**Pull request by thdaraujo**
### Motivation / Background

`#in_order_of` is an excellent method and makes ordering enum columns easier, and many complex order case/when queries can be refactored to use it.

This PR was created because I believe having more examples would help people understand this method a bit better.

Also, I think it's useful to document that this method will order results, but also FILTER them, excluding rows in which the column doesn't match any of the given values.

So I expanded the description and added these examples:
- what happens when dealing with `enum` +columns+
- what happens when passing `nil` as a +value+ for nullable columns

Here's a gist with some tests: https://gist.github.com/thdaraujo/7d9aa5ffd1b61bf57604ded14535e357

I'd be happy to add other examples for other usecases you think are relevant, just let me know!

### Detail

This Pull Request expands the documentation for [in_order_of](https://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-in_order_of)

### Additional information

n/a

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* ~[ ] Tests are added or updated if you fix a bug or add a feature.~ not applicable
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: activerecord/lib/active_record/relation/query_methods.rb [hunk 1]
```diff
@@ -606,7 +606,8 @@ def order!(*args) # :nodoc:
       self
     end
 
-    # Allows to specify an order by a specific set of values.
+    # Allows to specify an <code>ORDER BY</code> clause to a query based on a given +column+,
+    # ordered and filtered by the specific set of +values+.
```
Comment (p8) [inline activerecord/lib/active_record/relation/query_methods.rb:610]: I think the first sentence can be simplified a bit.
I'm not a fan of the `Allows to ..` as used by other methods in this file.
Removing it makes things simpler and clearer.

We also prefer `<tt>` for fixed width fonts:
https://edgeguides.rubyonrails.org/api_documentation_guidelines.html#fixed-width-font

I also don't think `in_order_of` filters the results.
```suggestion
    # Specify an <tt>ORDER BY</tt> clause to a query based on a given +column+,
    # ordered by a specific set of +values+.
```

I find "Specifying a clause to a query" a bit wordy still.
Maybe we can remove the "to a query" part as well.
```suggestion
    # Specify an <tt>ORDER BY</tt> clause based on a given +column+,
    # ordered by a specific set of +values+.
```

Or maybe making it similar to `order`:
```suggestion
    # Applies an <tt>ORDER BY</tt> clause to a query based on a given +column+,
    # ordered by a specific set of +values+.
```

File: activerecord/lib/active_record/relation/query_methods.rb [hunk 2]
```diff
@@ -617,6 +618,32 @@ def order!(*args) # :nodoc:
...
+    #   # SELECT "conversations".* FROM "conversations"
+    #   #   WHERE "conversations"."status" IN (1, 0)
+    #   #   ORDER BY CASE
+    #   #     WHEN "conversations"."status" = 1 THEN 1
+    #   #     WHEN "conversations"."status" = 0 THEN 2
+    #   #   END ASC
+    #
+    # +values+ can also include +nil+.
+    #
+    #   Conversation.in_order_of(:status, %w(nil archived active))
```
Comment (p8) [inline activerecord/lib/active_record/relation/query_methods.rb:638]: Not sure `nil` is working here as expected (it returns a String):
```ruby
%w(nil archived active)
=> ["nil", "archived", "active"]
```

**Review by p8 — CHANGES_REQUESTED** [review]

File: activerecord/lib/active_record/relation/query_methods.rb (hunk 1 above)
Comment (p8) [reply activerecord/lib/active_record/relation/query_methods.rb:610]: My bad, I didn't know it filters as well.

Comment (thdaraujo) [reply activerecord/lib/active_record/relation/query_methods.rb:610]: yeah, the filtering behavior was a bit surprising to me as well. 😄

File: activerecord/lib/active_record/relation/query_methods.rb (hunk 2 above)
Comment (thdaraujo) [reply activerecord/lib/active_record/relation/query_methods.rb:638]: nice catch!

**Comment by thdaraujo** [comment]
Thanks for the review @p8 , I've updated the documentation, please take a look when you have some time.
I created[ this gist](https://gist.github.com/thdaraujo/7d9aa5ffd1b61bf57604ded14535e357) with some example tests to double check the generated queries.

File: activerecord/lib/active_record/relation/query_methods.rb (hunk 1 above)
Comment (thdaraujo) [reply activerecord/lib/active_record/relation/query_methods.rb]: updated!

**Review by MatheusRich — APPROVED** [review]

**Comment by thdaraujo** [comment]
thanks for the review @MatheusRich !
